### PR TITLE
Shape sorting game

### PR DESCRIPTION
**Pull request by dmclaughlin104**
Created a basic gameplay loop for a shape sorting game. Wooden blocks appear in a random position in the middle of the screen, while their respective 'holes' appear in random positions along the sides. When all shapes are sorted in their correct holes, the shapes and holes are shuffled into new positions. Shapes glow yellow while being interacted with, and red when they are resting in the wrong spot. When they're in more or less the right spot, they will automatically move to centre of the hole and slot through it.

### CODE REVIEW

File: Assets/Scripts/DragDropToDistance.cs [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+    private Vector3 currentScreenPosition;
+    private float targetZ;
+    private float cameraDifferential = 5.5f;
+    private Rigidbody objectRB;
+    private bool isDragging;
+
+    private float moveSpeed = 0.9f;
+    [SerializeField] private string targetTag;
+
+    private bool isMoving = false; // Flag to track whether movement is in progress
```
Comment (eoloughlin05) [inline Assets/Scripts/DragDropToDistance.cs:32]: Could we change this to 'isShapeMoving'?

File: Assets/Scripts/DragDropToDistance.cs [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    private bool isAtMiddle = false;//to check whether object has reached middle;
```
Comment (eoloughlin05) [inline Assets/Scripts/DragDropToDistance.cs:33]: Could we change this to 'isShapeAtMiddleOfShapeSpace' or something indicating where the middle is?

File: Assets/Scripts/DragDropToDistance.cs [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        //once middle is reached, move backwards
+        if (isAtMiddle)
+        {
+            MoveBackWard();
+        }
+
+    }
+
+    //move object backward and through hole
+    private void MoveBackWard()
```
Comment (eoloughlin05) [inline Assets/Scripts/DragDropToDistance.cs:152]: We can change the method name of this to 'MoveObjectThroughHole' which would remove the comment line above.

File: Assets/Scripts/DragDropToDistance.cs [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+
+        //turning RB back on
+        TurnOnRB();
+
+    }
+
+    //methods to turn off/on the Rigid Body
+    private void TurnOffRB()
+    {
+        //GetComponent<Rigidbody>().useGravity = false;
```
Comment (eoloughlin05) [inline Assets/Scripts/DragDropToDistance.cs:198]: This comment can be removed if no longer required.

File: Assets/Scripts/DragDropToDistance.cs [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+        GetComponent<BoxCollider>().enabled = false;
+    }
+    private void TurnOnRB()
+    {
+        //droping object - turn RB back on
+        //GetComponent<Rigidbody>().useGravity = true;
```
Comment (eoloughlin05) [inline Assets/Scripts/DragDropToDistance.cs:204]: This comment can be removed if it is no longer required.

File: Assets/Scripts/DragDropToDistance.cs [hunk 6]
```diff
@@ -0,0 +1,268 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.InputSystem;
+
+public class DragDropToDistance : MonoBehaviour
+{
+    //followed this tutorial for basic touchscreen dragging: https://www.youtube.com/watch?v=zo1dkYfIJVg
```
Comment (eoloughlin05) [inline Assets/Scripts/DragDropToDistance.cs:10]: Can this line be removed?

File: Assets/Scripts/DragDropToDistance.cs [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+
+    private float moveSpeed = 0.9f;
+    [SerializeField] private string targetTag;
+
+    private bool isMoving = false; // Flag to track whether movement is in progress
+    private bool isAtMiddle = false;//to check whether object has reached middle;
+    private Transform targetTransform; // Reference to the transform of the target object
+
+
+
```
Comment (eoloughlin05) [inline Assets/Scripts/DragDropToDistance.cs:37]: Lines 35 - 37 can be removed as it's just white space.

File: Assets/Scripts/DragDropToDistance.cs [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+        //getting GameManager
+        gameManagerScript = GameObject.Find("GameManager").GetComponent<ShapeSorting_GameManager>();
+
+        //setting the target point in the Z axis for objects to be dragged into
+        playerCamera = Camera.main;
+        targetZ = playerCamera.transform.position.z + cameraDifferential;
+
+        //getting rigid body component
+        objectRB = GetComponent<Rigidbody>();
+
```
Comment (eoloughlin05) [inline Assets/Scripts/DragDropToDistance.cs:75]: This line can be removed too - re. white space

File: Assets/Scripts/DragDropToDistance.cs [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+        screenPosition.Enable();
+        press.Enable();
+
+        //to allow you to assess where on the screen has been clicked (I think)
+        screenPosition.performed += context => { currentScreenPosition = context.ReadValue<Vector2>(); };
+
+        //declaring what should happen press interaction starts
+        press.performed += _ =>
+        {
+
```
Comment (eoloughlin05) [inline Assets/Scripts/DragDropToDistance.cs:90]: This line can be removed - white space

File: Assets/Scripts/DragDropToDistance.cs [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+            if (isClickedOn)
+            {
+                StartCoroutine(Drag());
+            }
+
```
Comment (eoloughlin05) [inline Assets/Scripts/DragDropToDistance.cs:95]: This line can be removed - white space

File: Assets/Scripts/RotatingCubesController.cs [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+    /*
+    [SerializeField] InputAction press;
+    private Vector2 touchStartPos;
+    //private bool isTouching = false;
+    [SerializeField] Camera playerCamera;
+    private Vector3 currentScreenPosition;
+    */
+
+
```
Comment (eoloughlin05) [inline Assets/Scripts/RotatingCubesController.cs:18]: These lines can be removed if the code isn't required anymore.

File: Assets/Scripts/RotatingCubesController.cs [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+    {
+        if (_isRotating)
+        {
+            // offset
+            _mouseOffset = (Input.mousePosition - _mouseReference);
+
+            // apply rotation
+            _rotation.y = -(_mouseOffset.x + _mouseOffset.y) * _sensitivity;
+            // apply rotation
+            //_rotation.x = -(_mouseOffset.x + _mouseOffset.y) * _sensitivity;
```
Comment (eoloughlin05) [inline Assets/Scripts/RotatingCubesController.cs:41]: These lines can be removed if they're no longer required

File: Assets/Scripts/RotatingCubesController.cs [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+        press.Enable();
+
+        press.performed += _ =>
+        {
+            
+        };
+
+    }
+    */
+
```
Comment (eoloughlin05) [inline Assets/Scripts/RotatingCubesController.cs:85]: These lines can be removed if they're not required.

File: Assets/Scripts/ShapeSorting_GameManager.cs [hunk 14]
```diff
@@ -0,0 +1,160 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ShapeSorting_GameManager : MonoBehaviour
+{
+
```
Comment (eoloughlin05) [inline Assets/Scripts/ShapeSorting_GameManager.cs:7]: White space that can be removed

File: Assets/Scripts/ShapeSorting_GameManager.cs [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+    [SerializeField] private GameObject[] shapeHoles = new GameObject[6];
+    [SerializeField] private GameObject[] shapeObjects = new GameObject[6];
+    private Rigidbody[] objectRBs = new Rigidbody[6];
+
+    [HideInInspector] public int shapesInScene = 6;
+
+    public bool shapeReturning = false;
+    public int shapeToReturn;
+
+
```
Comment (eoloughlin05) [inline Assets/Scripts/ShapeSorting_GameManager.cs:19]: Extra line can be removed - white space

File: Assets/Scripts/ShapeSorting_PerimeterManager.cs [hunk 16]
```diff
@@ -0,0 +1,67 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ShapeSorting_PerimeterManager : MonoBehaviour
+{
+
```
Comment (eoloughlin05) [inline Assets/Scripts/ShapeSorting_PerimeterManager.cs:7]: White space - can be removed

File: Assets/Scripts/ShapeSorting_PerimeterManager.cs [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+        gameManagerScript = GameObject.Find("GameManager").GetComponent<ShapeSorting_GameManager>();
+
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
```
Comment (eoloughlin05) [inline Assets/Scripts/ShapeSorting_PerimeterManager.cs:26]: These lines can be removed as the Update method is empty and not in use.

**Review by eoloughlin05 — CHANGES_REQUESTED** [review]
I have tested this out Damien and it looks great. Brilliant work!

Just a few comments; very minor changes such as changing a couple of the naming conventions so that it is clearer in the code what is happening. There are a couple of unused or empty methods that can be removed and some comment lines also which I have marked. The last thing is white space; where there are a lot of extra blank lines that can be removed. I have marked a few of these, but not them all, just to give you an example of where I'm talking about. If there's anything that doesn't make sense, please just let me know. I will review it again as soon as changes are made. Thanks!

**Comment by dmclaughlin104** [comment]
Thanks for this, Erin.

I've now updated the scripts to remove blank space and I've also deleted a few older scripts that weren't currently in use or which have been superseded (including the RotatingCubes script). I've also made the red wrong highlight more obvious now by making the whole front of the shape red.

I'm up for adding sound effects etc., but I think I they should be in-keeping with the new sounds that we've recorded for the colour-sorting game.

In terms of priorities, I'll probably need to look at bringing in the sound effects into the colour-sorting game first, and then I look at doing the same for the shape-sorting game.

**Review by eoloughlin05 — APPROVED** [review]

**Pull request merged**